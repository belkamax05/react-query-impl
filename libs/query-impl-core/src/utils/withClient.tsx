import { QueryClient } from '@tanstack/query-core';
import { Hydrate, dehydrate } from '@tanstack/react-query';
import { AnyObject } from '../types';
import { AnyFunction } from '../types/AnyFunction';

const withClient = (fn: AnyFunction) => async (props: AnyObject) => {
  const queryClient = new QueryClient();
  const content = await fn({ ...props, queryClient });
  return <Hydrate state={dehydrate(queryClient)}>{content}</Hydrate>;
};

export default withClient;
