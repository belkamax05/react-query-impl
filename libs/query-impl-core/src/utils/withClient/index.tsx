import { QueryClient } from '@tanstack/query-core';
import { Hydrate, dehydrate } from '@tanstack/react-query';
import { AnyObject } from '../../types';
import { AnyFunction } from '../../types/AnyFunction';

const withClient = (fn: AnyFunction, queryClient?: QueryClient) => {
  return async (props: AnyObject) => {
    const queryClientInner = queryClient ?? new QueryClient();
    const content = await fn({ ...props, queryClient: queryClientInner });
    return <Hydrate state={dehydrate(queryClientInner)}>{content}</Hydrate>;
  };
};

export default withClient;
