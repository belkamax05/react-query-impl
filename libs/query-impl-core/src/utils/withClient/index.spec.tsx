import { QueryClient } from '@tanstack/query-core';
import withClient from '.';

describe('withClient', () => {
  const fn = jest.fn();
  fn.mockImplementation(({ queryClient }: { queryClient: QueryClient }) => {
    return <span>{queryClient.getQueryData(['test'])}</span>;
  });
  it('should pass queryClient to server component with default queryClient', async () => {
    const queryClient = new QueryClient();
    const Wrapped = withClient(fn);
    const resultWrapped = await Wrapped({ queryClient });

    expect(fn).toHaveBeenCalled();

    expect(resultWrapped.props.children).toMatchSnapshot();
  });

  it('should pass queryClient to server component with prop queryClient', async () => {
    const queryClient = new QueryClient();
    queryClient.setQueryData(['test'], 'test data');
    const Wrapped = withClient(fn, queryClient);
    const resultWrapped = await Wrapped({ queryClient });

    expect(fn).toHaveBeenCalled();

    expect(resultWrapped.props.children).toMatchSnapshot();
  });
});
