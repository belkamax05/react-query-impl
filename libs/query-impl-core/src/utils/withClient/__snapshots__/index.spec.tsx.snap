// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withClient should pass queryClient to server component with default queryClient 1`] = `<span />`;

exports[`withClient should pass queryClient to server component with prop queryClient 1`] = `
<span>
  test data
</span>
`;
